src/main/java/appeng/me/storage/SecurityStationInventory.java

```diff
@@ -70,13 +69,13 @@ public IAEItemStack injectItems( final IAEItemStack input, final Actionable type
 		return input;
 	}
 
-	private boolean hasPermission( final BaseActionSource src )
+	private boolean hasPermission( final IActionSource src )
 	{
-		if( src.isPlayer() )
+		if( src.player().isPresent() )
 		{
 			try
 			{
-				return this.securityTile.getProxy().getSecurity().hasPermission( ( (PlayerSource) src ).player, SecurityPermissions.SECURITY );
+				return this.securityTile.getProxy().getSecurity().hasPermission( src.player().get(), SecurityPermissions.SECURITY );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3655)